spec/src/main/asciidoc/architecture.asciidoc

```diff
@@ -1347,5 +1352,15 @@ Emitter<String> emitter;
 ----
 
 Since the `@Channel("myChannel")` is used to produce messages, a consumer with the `@Incoming("myChannel")` should be specified to consume the messages transiting on the channel _myChannel_.
-An `IllegalStateException` will be thrown if no consumer is found by the time a message is emitted to the channel.
+
+== Assembly and validation
+
+When the application starts, the Reactive Messaging implementation:
+
+* connect the methods annotated with `@Incoming` and `@Outgoing,` `Emitters,` channels and connectors
+* verify the validity of the resulting graph
+
+If the implementation cannot find a matching _upstream_ or _downstream_ channel for one of the components, a `DeploymentException` must be thrown when the application starts.
```

Comment:
If the implementation cannot find a matching _upstream_ or _downstream_ channel for one of the components, the `DeploymentException` must be thrown on application starting.